kamu-core/src/infra/ingest/fetch_service.rs

```diff
@@ -433,59 +410,56 @@ impl FetchService {
 
         if matched_files.is_empty() {
-            return if let Some(cp) = old_checkpoint {
-                Ok(ExecutionResult {
-                    was_up_to_date: true,
-                    checkpoint: cp,
-                })
+            return if prev_source_state.is_some() {
+                Ok(FetchResult::UpToDate)
             } else {
                 Err(IngestError::not_found(&fglob.path, None))
             };
         }
 
         let (first_filename, first_path) = matched_files.pop().unwrap();
-        let res = Self::fetch_file(
-            &first_path,
-            fglob.event_time.as_ref(),
-            None,
-            target,
-            listener,
-        );
-
-        let event_time = match event_time_source {
+        let fetch_res = Self::fetch_file(&first_path, None, None, target_path, listener)?;
+        assert_matches!(fetch_res, FetchResult::Updated(_));
+
+        let source_event_time = match event_time_source {
             None => None,
             Some(src) => Some(Self::extract_event_time(&first_filename, &src)?),
         };
 
-        match res {
-            Ok(exec_res) => Ok(ExecutionResult {
-                was_up_to_date: exec_res.was_up_to_date,
-                checkpoint: FetchCheckpoint {
-                    last_fetched: exec_res.checkpoint.last_fetched,
-                    source_event_time: event_time,
-                    last_filename: Some(first_filename),
-                    has_more: !matched_files.is_empty(),
-                    etag: None,
-                    last_modified: None,
-                },
-            }),
-            Err(e) => Err(e),
-        }
+        Ok(FetchResult::Updated(FetchResultUpdated {
+            source_state: Some(PollingSourceState::ETag(first_filename)),
+            source_event_time,
+            has_more: !matched_files.is_empty(),
+        }))
     }
 
     // TODO: Validate event_time_source
     // TODO: Support event time from ctime/modtime
     fn fetch_file(
         path: &Path,
-        _event_time_source: Option<&EventTimeSource>,
-        old_checkpoint: Option<FetchCheckpoint>,
+        event_time_source: Option<&EventTimeSource>,
+        prev_source_state: Option<&PollingSourceState>,
         target_path: &Path,
         listener: &dyn FetchProgressListener,
-    ) -> Result<ExecutionResult<FetchCheckpoint>, IngestError> {
+    ) -> Result<FetchResult, IngestError> {
         use fs_extra::file::*;
 
         info!(path = ?path, "Ingesting file");
 
+        match event_time_source {
+            None | Some(EventTimeSource::FromMetadata) => (),
+            Some(src) => {
+                return Err(EventTimeSourceError::incompatible(format!(
+                    concat!(
+                        "File source only supports deriving event time from metadata ",
+                        "(file modification time), found: {:?}"
+                    ),
+                    src
+                ))
+                .into())
```

Comment:
Is this temporary only or permanent?
If permanent, I guess we should not have this field in `FilesGlob` part within the ODF protocol:
```
        "eventTime": {
          "$ref": "/schemas/EventTimeSource",
          "description": "Describes how event time is extracted from the source metadata."
        },
```

Reply:
File source only supports `EventTimeSource::FromMetadata` which is one of the enum variants. We could make ODF schema smarter to somehow specify which variants are compatible, but not remove the field as in future we might want to add more sources (e.g. modified time, creation time etc.)